crates/quickjs-wasm-rs/src/js_binding/value.rs

```diff
@@ -147,13 +149,60 @@ impl Value {
     }
 
     pub fn as_str(&self) -> Result<&str> {
+        let buffer = self.as_wtf8_str_buffer();
+        str::from_utf8(buffer).map_err(Into::into)
+    }
+
+    pub fn as_str_lossy(&self) -> std::borrow::Cow<str> {
+        let mut buffer = self.as_wtf8_str_buffer();
+        match str::from_utf8(buffer) {
+            Ok(valid) => Cow::Borrowed(valid),
+            Err(mut error) => {
+                let mut res = String::new();
+                loop {
+                    let (valid, after_valid) = buffer.split_at(error.valid_up_to());
+                    res.push_str(unsafe { str::from_utf8_unchecked(valid) });
+                    const REPLACEMENT_CHAR: char = '\u{FFFD}';
+                    res.push(REPLACEMENT_CHAR);
+
+                    // see https://simonsapin.github.io/wtf-8/#surrogate-byte-sequence
+                    let lone_surrogate =
+                        matches!(after_valid, [0xED, 0xA0..=0xBF, 0x80..=0xBF, ..]);
+                    // Rust's `Utf8Error` reports a lone surrogate as having an error_len of 1
+                    // which results in 3 replacement chars being inserted instead of 1. 1
```

Comment:
Do you know why this happens? By reading the documentation of `Utf8Error` this is not how I interpret this piece. 

https://doc.rust-lang.org/stable/std/str/struct.Utf8Error.html#method.error_len

> Decoding should resume after that sequence (after inserting a [U+FFFD REPLACEMENT CHARACTER](https://doc.rust-lang.org/stable/std/char/constant.REPLACEMENT_CHARACTER.html)) in case of lossy decoding.